zero/src/rpc/jerigon.rs

```diff
@@ -53,9 +121,76 @@ where
                 compact: hex::decode(block_witness.strip_prefix("0x").unwrap_or(&block_witness))
                     .context("invalid hex returned from call to eth_getWitness")?,
             }),
-            txn_info: tx_results.into_iter().map(|it| it.result).collect(),
+            txn_info,
             code_db: Default::default(),
         },
         other_data,
     })
 }
+
+/// Processes the transactions in the given block and updates the code db.
+pub async fn process_transactions<'i, I, ProviderT, TransportT>(
+    block: &Block,
+    provider: &ProviderT,
+    tx_traces: I,
+) -> anyhow::Result<Vec<Option<JumpDestTableWitness>>>
+where
+    ProviderT: Provider<TransportT>,
+    TransportT: Transport + Clone,
+    I: Iterator<Item = &'i BTreeMap<H160, TxnTrace>>,
+{
+    let futures = block
+        .transactions
+        .as_transactions()
+        .context("No transactions in block")?
+        .iter()
+        .zip(tx_traces)
+        .map(|(tx, tx_trace)| process_transaction(provider, tx, tx_trace))
+        .collect::<FuturesOrdered<_>>();
+
+    let vec_of_res = futures.collect::<Vec<_>>().await;
+    vec_of_res.into_iter().collect::<Result<Vec<_>, _>>()
+}
+
+/// Processes the transaction with the given transaction hash and updates the
+/// accounts state.
+pub async fn process_transaction<ProviderT, TransportT>(
+    provider: &ProviderT,
+    tx: &Transaction,
+    tx_trace: &BTreeMap<H160, TxnTrace>,
+) -> anyhow::Result<Option<JumpDestTableWitness>>
+where
+    ProviderT: Provider<TransportT>,
+    TransportT: Transport + Clone,
+{
+    let tx_traces = tx_trace
+        .iter()
+        .map(|(h, t)| (Address::from(h.to_fixed_bytes()), t.clone()))
```

Comment:
Cloning `TxnTrace` is not a good idea. Lot of overhead.
There could be 2 solutions:
* Define `generate_jumpdest_table` with:
```
tx_traces: impl Iterator<Item = (&'a Address, &'a TxnTrace)>,
```
* Convert ` tx_trace` H160 to Address when you initially create it.

Reply:
Confusing here is that we use both `type Address = H160` and `alloy_primitives::Address`. But anyway, conversion should be do so that there is no unnecessary cloning of data.